extensions/amp-social-share/1.0/social-share.js

```diff
@@ -126,11 +142,11 @@ function throwWarning(message) {
 /**
  * @param {!JsonObject} typeConfig
  * @param {?string} baseEndpoint
- * @param {!JsonObject} props
+ * @param {!JsonObject} params
+ * @param {!JsonObject} bindings
  * @return {?string}
  */
-function createEndpoint(typeConfig, baseEndpoint, props) {
-  const {'params': params, 'bindings': bindings} = props;
+function createEndpoint(typeConfig, baseEndpoint, params, bindings) {
```

Comment:
I think I found a good metaphor of what I don't like about bindings here. It's the 

```
endpoint.replace(new RegExp(binding, 'g'), combinedBindings[binding])
```

What is our justification of using this code here with all the uncertainties that it brings?

Reply:
So this part of the code does something pretty similar to the following:
`urlReplacements.expandUrlAsync(hrefWithVars, bindings)`
which was in the original amp component, so I figured it made sense to provide the functionality in this component.

The user has full control of the bindings, so they can control it how they want or choose not to use bindings at all.

Is your concern that we are giving the user too much power in this component?

Reply:
If so, then we can completely remove the bindings functionality from the preact component.

Reply:
I have a few concerns:
1. `new RegExp(binding, 'g')` is ambiguous. Two bindings `ABC` and `ABC_TWO` will be matched by the same `ABC` and would give wrong results depending on the order of bindings.
2. I don't see why a component user would want this when they can simply pass all the right data they need directly.
3. This method would be hard to document.

So, IMHO, AMP way is likely necessary for AMP itself and that logic is reliant on the async resolution and probably it's the only way that can be done. But the component does not seem to need any of those features. And even if we wanted to preserve some of these features, they'd need to look quite different.

Reply:
Sounds good, I understand the concern, in that case let's remove the binding functionality completely from the Preact component and it will receive Params only.  I'll send an updated PR with this change.  Thanks for the input!